examples/gsm8k/gsm8k.py

```diff
@@ -0,0 +1,350 @@
+
+    def improve_prompt(self, **kwargs) -> str:
+        pass
+
+    def validation_prompt(self, **kwargs) -> str:
+        pass
+    
+    
+class GSM8KParser(parser.Parser):
+    """
+    GSM8KParser provides the parsing of the language model responses specific to the gsm8k example.
+
+    Inherits from the Parser class and implements its abstract methods.
+    """
+
+    def __init__(self) -> None:
+        """
+        Inits the response cache.
+        """
+        self.cache = {}
+
+    def parse_generate_answer(self, state: Dict, texts: List[str]) -> Dict:
+        """
+        Parse the response from the language model for a generate prompt.
+
+        :param state: The thought state used to generate the prompt.
+        :type state: Dict
+        :param texts: The responses to the prompt from the language model.
+        :type texts: List[str]
+        :return: The new thought states after parsing the response from the language model.
+        :rtype: List[Dict]
+        """
+        new_states = []
+        for text in texts:
+            if state["method"].startswith("io") or state["method"].startswith("cot"):
+                int_answer = utils.strip_int_result(text, state["method"])
+                if int_answer is not None:
+                    logging.warning(
+                        f"Could not parse step answer: {text}. Returning None."
+                    )
+                new_state = state.copy()
+                new_state["current"] = int_answer
+                new_state["phase"] = 2
+                new_states.append(new_state)
+            else:
+                raise ValueError(f"Unknown method: {state['method']}")
+        return new_states
+    
+    def parse_aggregation_answer(self, response: str, **kwargs) -> Union[Dict, List[Dict]]:
+        pass
+
+    def parse_improve_answer(self, response: str, **kwargs) -> Dict:
+        pass
+
+    def parse_validation_answer(self, response: str, **kwargs) -> bool:
+        pass
+
+    def parse_score_answer(self, states: List[Dict], texts: List[str]) -> List[float]:
+        assert len(states) == 1, "Scoring multiple states is not implemented."
+        score = []
+        for text in texts:
+            if "True" in text and not "False" in text:
+                score.append(1.0)
+            else:
+                score.append(0.0)
+        return score
+
+
+def io() -> operations.GraphOfOperations:
+    """
+    Generates the Graph of Operations for the IO method.
+
+    :return: Graph of Operations
+    :rtype: GraphOfOperations
+    """
+    operations_graph = operations.GraphOfOperations()
+
+    operations_graph.append_operation(operations.Generate(1, 1))
+    operations_graph.append_operation(operations.Score(1, False))
+    operations_graph.append_operation(operations.GroundTruth(utils.test_answer))
+
+    return operations_graph
+
+def cot() -> operations.GraphOfOperations:
+    return io()
+
+def cotsc() -> operations.GraphOfOperations:
```

Comment:
I think it'd be nice if we have a uniform naming scheme (but this is something we can do in the next iteration, since its only a minor thing)

Reply:
yes i will keep that comment unresolved